typing/jkind.ml

```diff
@@ -432,40 +438,122 @@ module Mod_bounds = struct
       nullability
     }
 
-  let join =
-    Map2.f
-      { f =
-          (fun (type axis) ~(axis : axis Axis.t) ->
-            let (module Bound_ops) = Axis.get axis in
-            Bound_ops.join)
-      }
+  let join
+      { locality;
+        linearity;
+        uniqueness;
+        portability;
+        contention;
+        yielding;
+        externality;
+        nullability
+      } t2 =
+    let locality = Locality.Const.join locality t2.locality in
+    let linearity = Linearity.Const.join linearity t2.linearity in
+    let uniqueness = Uniqueness.Const_op.join uniqueness t2.uniqueness in
```

Comment:
I'm very worried about these subtle `_op`s. You and I both wrote a few bugs in the past couple days due to getting them backwards. Maybe we should do some sort of aliasing so that there's more consistency between axes?

Reply:
Maybe make them all `Jkind_axis.<axis-name>`?